java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -69,4 +73,106 @@ public Optional<String[]> getColumns() {
   public long getBatchSize() {
     return batchSize;
   }
+
+  private ByteBuffer getProjection() {
```

Comment:
`getProjection` and `getFilter` should return optional values similar to `getColumns` if we remove `getSubstraitExtendedExpression` (see below).

Reply:
Changed